source/portable/os/ota_os_posix.c

```diff
@@ -27,7 +27,10 @@
 #include <stdlib.h>
 #include <string.h>
 #include <time.h>
-/* coverity[misra_c_2012_rule_21_5_violation] This is on linux. */
+
+/* MISRA rule 21.5 prohibits use of signal.h on embedded system. But this implementation is on linux.
+ * So it's a false positive. */
```

Comment:
I don't see anything that says this rule is only for embedded systems.  Can you provide a reference?

Reply:
I believe MISRA itself is for embedded system, quote from themselves,

MISRA, originally created as the Motor Industry Software Reliability Association, is a consortium concerned with best practice in developing safety- and security-related embedded systems.

Reply:
What about embedded linux systems?

Reply:
MISRA is really a set of rules to improve the quality of code written in C, it does not only apply to embedded code, it applies to all C-code. Technically standards like ISO 26262 and IEC 61508 require C code written as part of any critical system use a "safe subseet" of the language. This requirement is met by following the MISRA rules. If the system is written in C and runs on Linux this would equally apply.